run-e2e.sh

```diff
@@ -62,21 +65,25 @@ docker-compose up createtables
 docker-compose up init
 docker-compose up -d index relay
 sleep 3
-docker-compose logs -t -f parity index relay e2e &
 
-if test $use_local_yarn -eq 0; then
-    docker-compose up -d e2e
-    result=$(docker wait e2e)
-    docker-compose logs -t e2e >$mydir/output.txt
-    cd $cwd
-    if test $coverage -eq 1; then
-        rm -rf "$NYC_OUTPUT_DIR"
-        docker cp e2e:/clientlib/"$NYC_OUTPUT_DIR"/. "$NYC_OUTPUT_DIR"
+if test $only_backend -eq 0; then
+    docker-compose logs -t -f parity index relay e2e &
+    if test $use_local_yarn -eq 0; then
+        docker-compose up -d e2e
+        result=$(docker wait e2e)
+        docker-compose logs -t e2e >$mydir/output.txt
+        cd $cwd
+        if test $coverage -eq 1; then
+            rm -rf "$NYC_OUTPUT_DIR"
+            docker cp e2e:/clientlib/"$NYC_OUTPUT_DIR"/. "$NYC_OUTPUT_DIR"
+        fi
+    else
+        cd $cwd
+        yarn run test:e2e | tee $mydir/output.txt
+        result="${PIPESTATUS[0]}"
     fi
+    cat $mydir/output.txt
+    exit $result
 else
-    cd $cwd
-    yarn run test:e2e | tee $mydir/output.txt
-    result="${PIPESTATUS[0]}"
+    docker-compose logs -t -f parity index relay
```

Comment:
Yes makes definitely sense. How do I do that correctly? If I write `echo ...` directly into the bash script the output is not ideal because the command is logged as well. Do I have to create a service in the `docker-compose` file or how can I do this?

Reply:
'set -x' turns on the logging of the executed commands. I think we could turn that off 'set +x'.  But that would be annoying to do for every echo (we may add later).

stackoverflow to the rescue, I've implemented a solution based on DEBUG traps in PR #6
please review and merge it together with your stuff